frontend_tests/node_tests/voting_widget.js

```diff
@@ -76,4 +84,291 @@ run_test('poll_data_holder my question', () => {
+    const poll_comment_input = $.create('input.poll-comment');
+    const widget_comment_container = $.create('ul.poll-widget');
+
+    const poll_question = $.create('button.poll-question');
+    const poll_question_input = $.create('input.poll-question');
+    const poll_question_header = $.create('.poll-question-header');
+    const poll_question_container = $.create('.poll-question-bar');
+
+    const poll_vote_button = $.create('button.poll-vote');
+
+    widget_elem.set_find_results('button.poll-comment', poll_comment);
+    widget_elem.set_find_results('input.poll-comment', poll_comment_input);
+    widget_elem.set_find_results('button.poll-question', poll_question);
+    widget_elem.set_find_results('input.poll-question', poll_question_input);
+    widget_elem.set_find_results('.poll-question-header', poll_question_header);
+    widget_elem.set_find_results('ul.poll-widget', widget_comment_container);
+    widget_elem.set_find_results('button.poll-vote', poll_vote_button);
+    widget_elem.set_find_results('.poll-question-bar', poll_question_container);
+
+    poll_comment.on = (event, func) => {
+        assert.equal(event, 'click');
+
+        const e = {
+            stopPropagation: noop,
+        };
+
+        poll_comment_input.val('cool choice');
+        out_data = undefined;
+        func(e);
+        assert.deepEqual(out_data,  { type: 'new_comment', idx: 1, comment: 'cool choice' });
+
+        poll_comment_input.val('');
+        out_data = undefined;
+        func(e);
+        assert.deepEqual(out_data, undefined);
+    };
+
+    let poll_question_update_function;
+    poll_question.on = (event, func) => {
+        assert.equal(event, 'click');
+        poll_question_update_function = func;
+    };
+
+    poll_vote_button.on = noop;
+
+    voting_widget.activate(opts);
+
+    assert.equal(widget_elem.html(), 'poll-widget');
+    assert.equal(widget_comment_container.html(), 'poll-widget-results');
+    assert.equal(poll_question_header.text(), '');
+
+    const vote_events = [
+        {
+            sender_id: 100,
+            data: {
+                type: 'new_comment',
+                idx: 1,
+                comment: 'release now',
+            },
+        },
+        {
+            sender_id: 100,
+            data: {
+                type: 'vote',
+                key: '100,1',
+                vote: 1,
+            },
+        },
+    ];
+
+    poll_vote_button.on = (event, func) => {
+        assert.equal(event, 'click');
+
+        poll_vote_button.attr('data-key', '100,1');
+        const e = {
+            stopPropagation: noop,
+            target: poll_vote_button,
+        };
+        out_data = undefined;
+        func(e);
+        assert.deepEqual(out_data, { type: 'vote', key: '100,1', vote: 1 });
+    };
+
+    widget_elem.handle_events(vote_events);
+
+    assert(poll_question.attr('disabled'));
+    assert(poll_question_input.attr('disabled'));
+
+    const add_ques_event = [
```

Comment:
Go ahead and spell out "question."  I think "ques" is a fairly obscure abbreviation (unlike "ans", which is more common for some reason.)